src/formatDate.js

```diff
@@ -50,7 +50,63 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const TYPE_OF_YEAR_LONG = 'YYYY';
+  const TYPE_OF_YEAR_SHORT = 'YY';
+  const TYPE_OF_MONTH = 'MM';
+  const TYPE_OF_DAY = 'DD';
+  const CENTURE_21 = 20;
+  const CENTURE_20 = 19;
+  const LIMIT_YEAR = 30;
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const separatedDate = date.split(oldSeparator);
+  const newFormatDateArr = [];
+
+  function getIndex(arr, type) {
+    return arr.indexOf(type);
+  }
+
+  function findIndex(arr, type1, type2) {
+    return arr.findIndex(item => item.includes(type1) || item.includes(type2));
+  }
```

Comment:
Don't create helper functions inside another function, let's move them outside.
Also, getIndex looks not so useful